src/sync/update-leaderboard.js

```diff
@@ -51,44 +88,21 @@ const updateLeaderboardWithdrawing = async (syncPromises, events, limit) => {
             filtered.push(tx);
           }
         });
-        // calculate winnings for all participants
-        const promises = [];
-        each(filtered, (tx) => {
-          promises.push(new Promise(async (resolve, reject) => {
-            try {
-              const { eventAddress, betterAddress, resultIndex, centralizedOracleAddress } = tx;
-              const userAddress = betterAddress || centralizedOracleAddress;
-              // calculate winning for this user in this event
-              let winnings = '0';
-              if (resultIndex === eventObj.currentResultIndex) {
-                const res = await MultipleResultsEventApi.calculateWinnings({
-                  eventAddress,
-                  address: userAddress,
-                });
-                winnings = res.toString(10);
-              }
-              // update event leaderboard winnings
-              const eventLeaderboardEntry = new EventLeaderboard({
-                eventAddress,
-                userAddress,
-                investments: '0', // event leaderboard entry already has user's investments
-                winnings,
-              });
-              await DBHelper.insertEventLeaderboard(eventLeaderboardEntry);
-              // update global leaderboard investments, winnings
-              const globalLeaderboard = new GlobalLeaderboard({
-                userAddress,
-                investments: investments[userAddress],
-                winnings,
-              });
-              await DBHelper.insertGlobalLeaderboard(globalLeaderboard);
-              resolve();
-            } catch (err) {
-              reject(err);
-            }
-          }));
+        async.forEachOf(filtered, (value, key, callback) => {
```

Comment:
forgot to change to use `eachOf`. Will change and test later.

Reply:
please change before this gets merged, `eachOf` still is doing parallel exec.